### PR TITLE
feat: Add new field non_os_package_support to GAR resource

### PR DESCRIPTION
**Pull request by dmurray-lacework**
Issue: https://lacework.atlassian.net/browse/ALLY-664

Description:
Add new field non_os_package_support for gar terraform resource

Signed-off-by: Darren Murray darren.murray@lacework.net

### CODE REVIEW

File: examples/resource_lacework_integration_gar/variables.tf [hunk 1]
```diff
@@ -18,3 +18,7 @@ variable "private_key" {
   type      = string
   sensitive = true
 }
+variable "non_os_packages" {
```
Comment (vatasha) [inline examples/resource_lacework_integration_gar/variables.tf:21]: nit: let's name the variable `non_os_package_support` to be consistent.

File: integration/integration.go [hunk 2]
```diff
@@ -57,6 +57,18 @@ func GetIntegrationName(result string) string {
 	return res.Data[0].Name
 }
 
+func GetContainerRegistryIntegration(result string) api.ContainerRegIntegration {
```
Comment (vatasha) [inline integration/integration.go:60]: q: why are we doing this? I'm concerned about us splitting a string and the string itself changing. This feels a little brittle.

File: integration/resource_lacework_integration_gar_test.go [hunk 3]
```diff
@@ -18,9 +18,10 @@ func TestIntegrationGARCreate(t *testing.T) {
 		terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
 			TerraformDir: "../examples/resource_lacework_integration_gar",
 			Vars: map[string]interface{}{
-				"client_id":      gcreds.ClientID,
```
Comment (vatasha) [inline integration/resource_lacework_integration_gar_test.go:21]: nit: I think underscores for field names is very pythonic and not idiomatic go. I would use camelcase and not snakecase.

File: lacework/resource_lacework_integration_gar.go [hunk 4]
```diff
@@ -192,11 +198,12 @@ func resourceLaceworkIntegrationGarCreate(d *schema.ResourceData, meta interface
 	data := api.NewContainerRegistry(d.Get("name").(string),
```
Comment (vatasha) [inline lacework/resource_lacework_integration_gar.go:198]: I noticed we're doing a lot of type assertions without changing for an error. I think we should be checking for errors for all of these assertions.

**Review by vatasha — CHANGES_REQUESTED** [review]
Nothing major here but I'd love to have a conversation about error handling before this gets merged.

File: examples/resource_lacework_integration_gar/variables.tf (hunk 1 above)
Comment (vatasha) [reply examples/resource_lacework_integration_gar/variables.tf:21]:
```suggestion
variable "non_os_package_support" {
```

File: examples/resource_lacework_integration_gar/main.tf [hunk 5]
```diff
@@ -17,6 +17,7 @@ resource "lacework_integration_gar" "example" {
     private_key_id = var.private_key_id
     private_key    = var.private_key
   }
+  non_os_package_support = var.non_os_packages
```
Comment (vatasha) [inline examples/resource_lacework_integration_gar/main.tf:20]:
```suggestion
  non_os_package_support = var.non_os_package_support
```

File: integration/resource_lacework_integration_gar_test.go [hunk 6]
```diff
@@ -18,9 +18,10 @@ func TestIntegrationGARCreate(t *testing.T) {
...
-				"client_email":   gcreds.ClientEmail,
-				"private_key_id": gcreds.PrivateKeyID,
+				"client_id":       gcreds.ClientID,
+				"client_email":    gcreds.ClientEmail,
+				"private_key_id":  gcreds.PrivateKeyID,
+				"non_os_packages": false,
```
Comment (vatasha) [inline integration/resource_lacework_integration_gar_test.go:24]:
```suggestion
				"non_os_package_support": false,
```

File: lacework/resource_lacework_integration_gar.go (hunk 4 above)
Comment (vatasha) [reply lacework/resource_lacework_integration_gar.go:198]: Update: Discussed this and understand that the schema validates the type so there is no need for checking errors on the type assertions.

**Review by vatasha — APPROVED** [review]

File: integration/resource_lacework_integration_gar_test.go (hunk 3 above)
Comment (dmurray-lacework) [reply integration/resource_lacework_integration_gar_test.go:21]: We're overriding the terraform variables here. https://www.terraform-best-practices.com/naming

File: integration/integration.go (hunk 2 above)
Comment (dmurray-lacework) [reply integration/integration.go:60]: It certainly could be improved. The string being split is the terraform output, which is consistent. If there was no output or no id in the output the tf step would have failed. But I can look at pattern matching instead.

**Review by afiune — APPROVED** [review]
tenor-157862798

**Comment by afiune** [comment]
@dmurray-lacework the pipeline is complaining about:
```
Executing command: make test                                                                                                                                          
==> Checking that code complies with gofmt requirements...                                                                                                            
gofmt needs running on the following files:                                                                                                                           
./integration/resource_lacework_integration_gar_test.go                                                                                                               
You can use the command: `make fmt` to reformat code.                                                                                                                 
make: *** [GNUmakefile:113: fmtcheck] Error 1                                                                                                                         
```

**Pull request merged**